presto-hive/src/main/java/com/facebook/presto/hive/HudiDirectoryLister.java

```diff
@@ -0,0 +1,135 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.hive;
+
+import com.facebook.airlift.log.Logger;
+import com.facebook.presto.hive.filesystem.ExtendedFileSystem;
+import com.facebook.presto.hive.metastore.Table;
+import com.facebook.presto.hive.util.HiveFileIterator;
+import com.facebook.presto.spi.ConnectorSession;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.BlockLocation;
+import org.apache.hadoop.fs.FileStatus;
+import org.apache.hadoop.fs.LocatedFileStatus;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.fs.PathFilter;
+import org.apache.hadoop.fs.RemoteIterator;
+import org.apache.hudi.common.config.HoodieMetadataConfig;
+import org.apache.hudi.common.engine.HoodieEngineContext;
+import org.apache.hudi.common.engine.HoodieLocalEngineContext;
+import org.apache.hudi.common.fs.FSUtils;
+import org.apache.hudi.common.model.HoodieBaseFile;
+import org.apache.hudi.common.table.HoodieTableMetaClient;
+import org.apache.hudi.common.table.view.FileSystemViewManager;
+import org.apache.hudi.common.table.view.HoodieTableFileSystemView;
+
+import java.io.IOException;
+import java.util.Iterator;
+import java.util.Optional;
+
+import static com.facebook.presto.hive.HiveFileInfo.createHiveFileInfo;
+import static com.facebook.presto.hive.HiveSessionProperties.isHudiMetadataEnabled;
+import static org.apache.hadoop.hdfs.DFSConfigKeys.DFS_DATANODE_DEFAULT_PORT;
+
+public class HudiDirectoryLister
+        implements DirectoryLister
+{
+    private static final Logger log = Logger.get(HudiDirectoryLister.class);
+
+    private final HoodieTableFileSystemView fileSystemView;
+    private final HoodieTableMetaClient metaClient;
+    private final boolean metadataEnabled;
+
+    public HudiDirectoryLister(Configuration conf, ConnectorSession session, Table table)
+    {
+        log.info("Using Hudi Directory Lister.");
+        this.metadataEnabled = isHudiMetadataEnabled(session);
+        this.metaClient = HoodieTableMetaClient.builder()
+                .setConf(conf)
+                .setBasePath(table.getStorage().getLocation())
+                .build();
+        HoodieEngineContext engineContext = new HoodieLocalEngineContext(conf);
+        HoodieMetadataConfig metadataConfig = HoodieMetadataConfig.newBuilder()
+                .enable(metadataEnabled)
+                .build();
+        this.fileSystemView = FileSystemViewManager.createInMemoryFileSystemView(engineContext, metaClient, metadataConfig);
+    }
+
+    public Optional<HoodieTableMetaClient> getMetaClient()
```

Comment:
Is this method an override ? If not, why does it return optional, though the underlying value is always present.

Reply:
Yeah, optional is not needed here. I will remove it.